code/stippler.py

```diff
@@ -0,0 +1,232 @@
+    parser.add_argument('--interactive', action='store_true',
+                        default=default["interactive"],
+                        help='Display intermediate results (slower)')
+    args = parser.parse_args()
+
+    filename = args.filename
+    density = 1.0 - scipy.misc.imread(filename, flatten=True, mode='L')/255.0
+    density = density[::-1, :]
+    density = normalize(density)
+
+    dirname = os.path.dirname(filename)
+    basename = (os.path.basename(filename).split('.'))[0]
+    pdf_filename = os.path.join(dirname, basename + "-stipple.pdf")
+    png_filename = os.path.join(dirname, basename + "-stipple.png")
+    dat_filename = os.path.join(dirname, basename + "-stipple.npy")
+
+    print("Density file: %s (%dx%d)" % (
+          filename, density.shape[1], density.shape[0]))
+
+    # Initialization
+    if not os.path.exists(dat_filename) or args.force:
+        points = initialization(args.n_point, density)
+        print("Nb points:", args.n_point)
+        print("Nb iterations:", args.n_iter)
+    else:
+        points = np.load(dat_filename)
+        print("Nb points:", len(points))
+        print("Nb iterations: -")
+        
+    xmin, xmax = 0, density.shape[1]
+    ymin, ymax = 0, density.shape[0]
+    bbox = np.array([xmin, xmax, ymin, ymax])
+    ratio = (xmax-xmin)/(ymax-ymin)
+
+    # Interactive display
+    if args.interactive:
+
+        # Setup figure
+        fig = plt.figure(figsize=(args.figsize, args.figsize/ratio),
+                         facecolor="white")
+        ax = fig.add_axes([0, 0, 1, 1], frameon=False)
+        ax.set_xlim([xmin, xmax])
+        ax.set_xticks([])
+        ax.set_ylim([ymin, ymax])
+        ax.set_yticks([])
+        scatter = ax.scatter(points[:, 0], points[:, 1], s=1, color=".25")
+
+        def update(frame):
+            global points
+            regions, points = voronoi.centroids(points, density)
+            Pi = points.astype(int)
+            X = np.maximum(np.minimum(Pi[:, 0], density.shape[1]-1), 0)
+            Y = np.maximum(np.minimum(Pi[:, 1], density.shape[0]-1), 0)
+            sizes = (args.pointsize[0] +
+                     (args.pointsize[1]-args.pointsize[0])*density[Y, X])
+            scatter.set_offsets(points)
+            scatter.set_sizes(sizes)
+            bar.update()
+
+            # Save stipple points and stippled image
+            if (frame == args.n_iter-1 and
+                    not os.path.exists(dat_filename) or args.save):
+                np.save(dat_filename, points)
+                plt.savefig(pdf_filename)
+                plt.savefig(png_filename)
+
+        bar = tqdm.tqdm(total=args.n_iter)
+        animation = FuncAnimation(fig, update,
+                                  repeat=False, frames=args.n_iter-1)
+        plt.show()
+
+    elif not os.path.exists(dat_filename) or args.force:
+        for i in tqdm.trange(args.n_iter):
+            regions, points = stipple(points, density)
```

Comment:
... and trying to put a filename at the end does not work: `stippler.py: error: unrecognized arguments: output.png`